operator/controllers/metrics/keptnmetric_controller.go

```diff
@@ -61,12 +73,61 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	return ctrl.Result{}, nil
+	if time.Now().Before(metric.Status.LastUpdated.Add(time.Second * time.Duration(metric.Spec.FetchIntervalSeconds))) {
+		r.Log.Info("Metric has not been updated for the configured interval. Skipping")
+		return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	evaluationProvider, err := r.fetchProvider(ctx, types.NamespacedName{Name: metric.Spec.Provider.Name, Namespace: metric.Namespace})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info(err.Error() + ", ignoring error since object must be deleted")
+			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+		r.Log.Error(err, "Failed to retrieve the provider")
+		return ctrl.Result{}, nil
+	}
+	// load the provider
+	provider, err2 := providers.NewProvider(metric.Spec.Provider.Name, r.Log, r.Client)
+	if err2 != nil {
+		//r.recordEvent("Error", metric, "ProviderNotFound", "provider was not found")
```

Comment:
why don't we record at least some Events here ?

Reply:
forgot I will adapt to your code